### ISSUE TITLE
Require user before serving static files

### ISSUE THREAD

**Issue by jstockwin** [keystonejs/keystone#2661]
Hi All, 

Firstly, this is all quite new to me so apologies if I am missing something obvious. 

I am trying to have a directory in which to upload files which should only be accessible to logged in users. Either by using the existing `/public` directory and adding a `/public/private` directory within this, or by adding a new `/private` static folder all together. 

I was hoping to achieve something like [this](http://stackoverflow.com/questions/9457511/redirecting-to-a-static-file-in-express-js), and add my own middleware in `/routes/middleware.js`, and then add `app.get('/private:file', middleware.newFunction);` into `routes/index.js`.

If using `/public/private` then it seems that the file is served anyway, and this code is never reached. (The console doesn't log `GET /private/...` for files in the public directory. 

When trying with a new `/private` folder, I was unsure how to proceed with setting a new express.static directory to the app. Obviously setting `'static': 'public private'` in `keystone.js` will result in the same problems as above, so it needs to be added after this. I tried instead adding `keystone.set('static', 'private');` at the end of `keystone.js`, after the routes had been set, but this didn't seem to change anything. 

The keystone docs also point out that you can set `static options` in `keystone.js`, and refer you [here](https://github.com/expressjs/serve-static), however this didn't seem to be quite the right thing. 

Any Ideas? All I want is a private folder somewhere, such that only logged in users can access the containing files. The files are handled by a LocalFiles model, so will change frequently, and thus it makes sense to set it as a static folder.

**Comment by geloescht** [comment]
You can add your own routes to express. Keystone already uses express but you are free to add other routes, including static routes, as you wish. You may also configure LocalFiles to store files in any directory.

**Comment by jstockwin** [comment]
Thanks. I had managed to add both additional static routes, and also store LocalFiles elsewhere. I guess what I really wanted to be able to do, was have a static route, but only after middleware.requireUser has passed. I think this is just not possible with how static routes work. They're not designed for you to also be able to run middleware before them.

I've managed achieve what I wanted without using static routes, so will close this issue. 

If anyone else is struggling with this, I just created a route ([link](https://github.com/Students4Students/KeystoneJS-Website/blob/master/routes/views/fileviewer.js)) which downloads the files I wanted. Then add this route to routes/index.js after the standard `middleware.requireUser`.
